### ISSUE TITLE
Have separate URL for different components

### ISSUE THREAD

**Issue by imatiach-msft** [tensorflow/tensorboard#997]
It would be nice to have separate URLs for different components - for example, I would like to be able to view just the tensorboard scalar graphs and not the rest of tensorboard in an iframe; or, I would like to see the layers visualization as a separate iframe.  Currently, it looks like this is not possible.  I would like to be able to view the components without the rest of the tensorboard UI.

**Comment by nfelt** [comment]
This would probably be a fairly substantial change but it's something we may want to do for structural reasons anyway.

When you mean "the rest of the tensorboard UI" I assume you're basically just referring to the orange navigation bar along the top?

**Comment by chihuahua** [comment]
Thanks. I'm curious what the motivation is.

Is the motivation related to being able to view individual dashboards in full-screen mode? ie, like Nick noted, would a simple toggle to hide distracting UI elements like the navigation bar suffice?

Many of TensorBoard's components are embedded within standalone web apps. For instance, the projector: http://projector.tensorflow.org/

**Comment by imatiach-msft** [comment]
@chihuahua yes, the motivation is to be able to view individual dashboards or even components of dashboards -- specifically, as widgets inside jupyter notebooks, but I could imagine other devs might want to leverage tensorboard in other ways.  Agreed, hiding distracting elements might suffice.

**Comment by jart** [comment]
I agree that this functionality should exist, and it'd be reasonably easy to do if we put all our raw sources in the webfiles.zip static assets file. Please note the iframe would load very slow, since the HTML wouldn't be vulcanized. I've been thinking about the idea of having vulcanization built in to our HTTP serving somehow, specifically due to use cases like these. It might not be something we have time to work on right this moment.

**Comment by nfelt** [comment]
cc @nsthorat @dsmilkov 

The projector.tensorflow.org page is less a "customization" or "extension" of TensorBoard than it is just taking the projector dashboard's main visualization and rendering it in a different frame.  The `vz-projector-app` polymer element here is an example of doing that:
https://github.com/tensorflow/tensorboard/blob/master/tensorboard/plugins/projector/vz_projector/vz-projector-app.html

**Comment by vmayoral** [comment]
Many thanks @nfelt

**Comment by bileschi** [comment]
@stephanwlee is there anything in our roadmap about being able to retrieve individual frames for plugins or for elements within those plugins?
